### DISCUSSION TITLE
Log tells bridged camera support is discontinued, does this effect Ring Door Bell cameras?

### DISCUSSION THREAD

**Discussion by woccam** [dgreif/ring#1489] [Q&A]
The log tells: "Bridged camera support will be removed in the next major release of homebridge-ring. Please enable the unbridgeCameras option in your configuration and add the individual cameras to HomeKit to prepare for this change." Does this affect Ring Door cameras? Does this mean they will no longer be supported at all or how do I integrate them directly?

**Comment by tsightler** [comment]
A Ring doorbell is a camera, so yes, they are impacted.  The unbridgeCameras option is not new, it's just that, in a future version, that will be the only option, so we are suggesting users go through the steps now.  It just means that, instead of adding the bridge once, and all the cameras show up, the cameras must be added as an individual accessories to Homekit.  The process of adding an individual camera is not really different than adding a bridge and is already discussed here in multiple threads.

Note that this is not something unique, if you purchased individual Homekit compatible cameras, you'd have to add those as individual cameras as well.  With this setting, each camera is treated as an individual camera which, underneath, has many benefits which impact maintainability and, in certain cases, usability, especially cases where a user has a large number of cameras.  Yes, it's somewhat of a pain to go through and add 8 individual cameras, but, overall, the cameras will likely work far better after this.

**Comment by scheckware** [comment]
@tsightler I changed my setting to "unbridgeCameras": true and I was able to add back my various Ring flood lights and a Ring doorbell that is shared to me from my son's Ring doorbell at his school, but my front door Ring doorbell keeps taking a long time and then eventually says "Unable to Add Accessory, Accessory is not reachable".    Any ideas what I can do?

**Reply by scheckware** [reply]
It seems that completely shutting down my phone and restarting and then trying again in the home app to add the Ring doorbell now immediately worked.    I tried it like 20 times force quitting the Home app and restarting home bridge and nothing worked but restarting the phone did it.

**Reply by gdavids57** [reply]
Just to add to the instructions... While adding the camera accessory in the Home app, the app will prompt for a Setup Code.  Use the Homebridge setup QR code on the Status page.  After completing the setup and seeing the camera in the Home app, I'm still seeing a message in the Homebridge log calling for a manual setup of the camera.  Is this normal.

**Reply by tsightler** [reply]
I wouldn't say the message is calling for anything to be done, the message is simply providing the information on how to add cameras (the message is unchanged for years).  The plugin doesn't have any way to know if you've already added the devices already or not.  Is there some other wording for that informational message that would make that more clear?

**Reply by gdavids57** [reply]
I think a clause like, "if you've made the changes, ignore the following..." would be helpful.

**Reply by tsightler** [reply]
I don't think that really statement really fits though as it's not always changes that need to be made, it could be initial setup for example and it needs to be on the same line as sometimes log messages are not in strict order.  Maybe it could just be reworded to sounds less like a request?  Homebridge logs the steps for adding itself to Homekit on every boot, regardless of whether it was already added or not, and somehow that doesn't seem to confuse people so it's not really clear to me why these log lines are unclear.  I think about possible tweaks to the phrasing.

**Comment by searls** [comment]
In response to this deprecation warning, I've tried taking my two Spotlight cameras onto unbridged mode, but HomeKit times out after adding either of them. Running on Node v20 with the latest version of the plugin, have rebooted Homebridge, my iPhone, and the cameras half a dozen times, but each time I eventually get Unable to Add Accessory:

![IMG_2075](https://github.com/user-attachments/assets/ba75f0f0-a657-4e31-97af-cde0f4aa59ce)

**Reply by tsightler** [reply]
Not reachable can mostly be two things:

1) Firewall -- Standalone accessories bind to separate ports randomly assigned by Homebridge during startup (the allowed port range can be configured in Homebridge config), these ports must be open on the host.
2) Incorrect mDNS advertisement -- Devices are automatically discovered via mDNS.  For hosts that are multi-homed, you should make sure that Homebridge is bound only to a single interface in the settings.  If you are sure there is no firewall related issue, and you are on the same network to which Homebridge is bound, then you might try changing the mDNS advertiser.  I've generally found Ciao to work well.

**Reply by searls** [reply]
Thanks @tsightler -- no firewall, one home, on Ciao.

See this kind of thing from time to time with HomeKit regardless, and I tend to blame Unifi (1 USG + 4 APs on one network/SSID) and in the past tinkering with mDNS-adjacent settings helps, but right now the other 150 devices in my house are actually working great and this is the only thing giving me trouble. Any Ubiquiti/Unifi specific advice?

**Reply by tsightler** [reply]
Apologies for being skeptical, but I've been supporting users with this for over 5 years and I no longer have enough fingers or toes to count the people that told me "no firewall" only for me to remote with them or whatever and find out they had Windows/Linux default firewall, or Synology with default firewall or whatever.

I'm also a Unifi user, never had a problem, ever, assuming you are using default settings.  I've found most users that have issues with Unifi because they change defaults without really understanding the impact, like Client Isolation (at least the modern Unifi UI tells users these will cause problems, but people still turn it on because it sounds "secure") or mDNS services (which are generally only needed for multi-subnets and, even then should be configured with understanding, not just turned on).  I'm not saying any of this is you specifically, just saying that my history hundreds of issues like this always make me skeptical.

In the end, the overall concept here is super simple, services are advertised via mDNS, and another service discovers those services, which includes the IP and port, and tries to connect to that.  There's really not a lot that can go wrong except mDNS and firewall/network.

Perhaps you can try using and mDNS discovery tool, something like Bonjour Browser, which will show you the list of mDNS devices on your network, including their name, capabilities, and also the IP/port they are advertising.  This should match the IP/port seen in the Homebridge logs.  If they do, then problem pretty much has to be reachability of that port from the device, if not, or if the devices isn't listed at all, the problem has to be with mDNS advertisement.

I can't really think of anything else, maybe devices with special characters which aren't allowed, but normally that would keep the device from appearing at all, in your case it seems to device appears, so it's in mDNS, but you can't connect to it.

**Reply by tsightler** [reply]
BTW, I've found Bonjour Brower to work reasonably well on my Windows systems, although you may have to refresh a few times, but not as well on Andriod, however, there's an app called Service Browser, which works pretty well on Andriod. Cameras configured as unbridged accessories will be listed under _hap.tcp., when you select it you can see the IP address and port it is advertising. When you select a device, it just tries to connect to that IP/port and, if it can't, then accessory is not reachable is the common error.

**Reply by searls** [reply]
Thanks for following up. Still stuck, but was interesting to explore a bit.

Using Discovery ([MAS link](https://apps.apple.com/us/app/discovery-dns-sd-browser/id1381004916?mt=12)), I could find the devices under HAP:
![screenshot-2024-12-15-08h44m18s](https://github.com/user-attachments/assets/a3657c3d-ef01-4382-b165-47b220afce9a)

Both these had 172… addresses, which I take to mean is "bridged" via Docker or some such in my Homebridge setup.

Then I restarted one of the cameras after toggling the "unbridge cameras option" and it showed a local IP address and port: 192.168.1.65:38684

That seems good!

Then I tried connecting via the command line, also good:

```
$ nc -zv 192.168.1.65 38684
Connection to 192.168.1.65 port 38684 [tcp/*] succeeded!
```

And (as before), Home app could see both cameras in the "Add Accessory" -> "More Options" UI, but even after restarting the camera and homebridge and my phone, I still get the failure to connect error.

THEN after the pairing failed, the camera disappeared from Bonjour. I restarted Homebridge and it reappeared but once again with an unreachable 172.17.0.1:33646 address.

Based on your experience does this sound like an mDNS/Bonjour issue or a Homebridge/plugin issue?

**Reply by tsightler** [reply]
It's a bit difficult to know since you haven't shared details of how you are running Homebridge, and you didn't provide logs showing what ports the accessories are registering on from the perspective of Homebridge, so all I can really do is guess, however, I'll give it a shot.

The fact that it continues to register on what appears to be a Docker bridge network IP on the host (just a guess, but it looks like the default IP of the docker bridge network on the host), which shouldn't happen, it's a Docker host configuration issue, for example, avahi-daemon running on the host, as, otherwise I have no idea how the Ciao instance in Homebridge could even know about that address, unless you are running Homebridge on the bridge network.

**Reply by tsightler** [reply]
Thinking about this a bit more, you said above that you are using Ciao.  One of the things different about Ciao vs Bonjour-HAP is that it, by default, advertises on all interfaces.  If you are running Homebridge in Docker using the host network, which is the recommended configuration, it's possible that, for some reason Ciao is finding the Docker bridge network and advertising there as well.  I'd make 100% sure that, in the Homebridge Network config that you have explicitly selected the single interface that you want to advertise on.

**Reply by searls** [reply]
Hmm, I'm running the official [Homebridge package on my Synology](https://github.com/homebridge/homebridge/wiki/Install-Homebridge-on-Synology-DSM), so I'm sure that's the root of a lot of that. Thanks for replying! 

Sorry for not providing more info earlier. I'm on the latest release version of homebridge and homebridge-ring, running under Node 20

These are the only relevant logs I found for the camera I'm focused on:
```
[12/15/2024, 3:21:41 PM] [homebridge-ring] Configured camera 631599d8-ef4f-4934-8b5e-09720605b593 stickup_cam_v4 Lanai
[12/15/2024, 3:21:41 PM] Publishing external accessory (name: Lanai, publishInfo: {
  username: '73:E6:B2:5F:57:D8',
  pincode: '***-**-***',
  category: 17,
  port: undefined,
  bind: undefined,
  mdns: undefined,
  addIdentifyingMaterial: true,
  advertiser: 'ciao'
}).
[12/15/2024, 3:21:41 PM] Lanai 50B8 is running on port 32886.
[12/15/2024, 3:21:41 PM] Please add [Lanai 50B8] manually in Home app. Setup Code: 786-68-022
```

**Reply by searls** [reply]
>  I'd make 100% sure that, in the Homebridge Network config that you have explicitly selected the single interface that you want to advertise on.

Thanks for this, I wasn't aware of this configuration at all. It looks like this wasn't configured:

![screenshot-2024-12-15-15h28m05s](https://github.com/user-attachments/assets/d170eeb7-ec4b-4b59-bb88-46b02d98f644)

Rebooting everything and trying again

**Reply by tsightler** [reply]
Yes, running HB on Synology is a bit challenging, especially if you are using any Mac services on the Synology, for example, to share files to Mac systems, because that likely means that the host itself is running Avahi daemon for mDNS.  In that world you'll almost certainly want to select Avahi for the advertiser because any other advertiser will just conflict.

You need to make sure that, whatever ports the accessories are advertising on, and whatever IP your Synology host is using, is what is always advertised.  It sounds like, for some reason, something keeps advertising these devices on the Docker network which will be unreachable outside of the Synology host itself.  If you can fix that it will probably work, assuming there is not firewall running on the Synology (the Synology does have a firewall built in, but is disabled by default, still might be worth a double-check).

If your Synology is running a firewall, you'll need to configure Homebridge to use a specific range of ports for accessories in HB config, and then make sure those ports are open in the firewall runs in **Control Panel > Security > Firewall**.

**Reply by searls** [reply]
Thanks again for the help in chasing this down. Unfortunately I had to travel unexpectedly so may not be able to follow up on this for a few weeks--will try all this when I get back!

**Reply by tsightler** [reply]
OK, let us know if you still need help when you get back to it, happy to continue to dig into it.  Save travels!

**Reply by Falter** [reply]
I'm in a similar boat. Thought I would 'Unbridge' and prep for the future release while I had some downtime and while the devices show up in Home to be added, I was getting the same "unable to connect error" when I try to add them. I'm running HomeBridge 1.8.5 (UI 4.65.1) with Node.js version 18.20.5 in a Rocky Linux 9.5 VM on ProxMox 8.3.1 (all patched current to today). There are no firewall rules set up for the VM in ProxMox, only 1 networking interface, and my phone is on the same Wifi network (Unifi, of course?) and neither device is locked to a specific AP or on different VLANs or subnets. I tried switching from Bonjour to Avahi (not supported errors) and then to Ciao with the same error persisting.

So, I upgraded to Node.js version v22.12.0 using the "sudo hb-service update-node" command. Now, the devices come up in logs saying they need to be setup - and they show in the Home App on my phone to add - but when I try any of the I get a "Connection Lost" error and they all disappear from the Discovery app linked above (with Ciao enabled). With Bonjour re-enabled, I am back to the original timeout error and things are no longer disappearing in Discovery when I try to add them.

Fortunately, when I re-enable bridging, everything pops right back up in Home.
Any suggestions on what stupid mistake I have made, by chance? Thank you!

**Reply by tsightler** [reply]
The troubleshooting steps are the same,:

1) Use some tool to verify that the services are advertised with correct IP/port (i.e. match what is in the Homebridge logs)
2) Verify that some device on the wireless network can connect to those ports with a TCP port test

You said "There are no firewall rules set up for the VM in ProxMox" but Rocky Linux installs and configures a firewall (firewalld) by default, so are you saying you disabled it?  Running `sudo systemctl status firewalld` shows the local OS firewall is inactive?

**Reply by Falter** [reply]
Thank you for your help!
It was Rocky's firewall. Dagnabbit!

systemctl stop firewalld
systemctl disable firewalld

Everything is working as expected now.

It looks like Homebridge makes an entry for its port in /etc/firewalld/zones/public.xml - I guess that would be why bridging was working.

If you don't mind my asking, does the driver assign the port numbers for the Ring cameras or does each device choose a port and then the driver advertises that port with the Homebridge IP? I wonder if a range could be reserved so a firewall entry could be created, for example?

Either way, I thank you for your help and made a donation to the project. If I can support you directly for your help, please let me know. Someone needs to be there to tell idgits like me how to find our firewalls - you're doing God's work, sir!
